module/os/linux/zfs/zfs_acl.c

```diff
@@ -1570,7 +1570,7 @@ zfs_acl_chmod(zfsvfs_t *zfsvfs, uint64_t mode, zfs_acl_t *aclp)
 	list_insert_tail(&aclp->z_acl, newnode);
 }
 
-void
+int
```

Comment:
I see FreeBSD includes code to handle this error, let's add the missing error handling from the caller while we're at it on the Linux side even if it can't currently fail.  The same error path can be used.

```c
zfs_setattr(vnode_t *vp, vattr_t *vap, int flags, cred_t *cr,
        <snip>
                if (err = zfs_acl_chmod_setattr(zp, &aclp, new_mode))
                        goto out;
```